pkg/kubelet/images/image_manager.go

```diff
@@ -164,17 +243,89 @@ func (m *imageManager) EnsureImageExists(ctx context.Context, pod *v1.Pod, conta
 		m.backOff.Next(backOffKey, m.backOff.Clock.Now())
 
 		msg, err := evalCRIPullErr(container, imagePullResult.err)
-		return "", msg, err
+		return imageRef, msg, err
 	}
 	m.podPullingTimeRecorder.RecordImageFinishedPulling(pod.UID)
 	imagePullDuration := time.Since(startTime).Truncate(time.Millisecond)
 	m.logIt(ref, v1.EventTypeNormal, events.PulledImage, logPrefix, fmt.Sprintf("Successfully pulled image %q in %v (%v including waiting). Image size: %v bytes.",
 		container.Image, imagePullResult.pullDuration.Truncate(time.Millisecond), imagePullDuration, imagePullResult.imageSize), klog.Info)
 	metrics.ImagePullDuration.WithLabelValues(metrics.GetImageSizeBucket(imagePullResult.imageSize)).Observe(imagePullDuration.Seconds())
 	m.backOff.GC()
+
+	if m.pullImageSecretRecheck {
+		m.lock.Lock()
+		defer m.lock.Unlock()
+		// Get the ImageRef after the Image Pull
```

Comment:
Actually, we are not pulling the image here ,instead we perform the Get operation where it returns the image reference in case if the image is present locally on the machine.
This info is required just to decide if we have to perform a Pull again or not. So, we need not add mechanism to record any events related to pull right?
Am I missing something here?

Reply:
Good point @liggitt, keeping a list of in progress pull attempts is nec. to manage a disconnect between 2 and 3..